backend/user.js

```diff
@@ -191,47 +208,161 @@ router.get('/myprofile', authenticateToken, async (req, res) => {
+      }
+    }
+
+    return res.status(200).json(deletedProfile);
   } catch (error) {
-    console.error('Error fetching patient profile:', error);
-    res.status(500).json({ error: 'Failed to fetch patient profile' });
+    console.error('Error deleting patient profile:', error);
+    res.status(500).json({ error: 'Failed to delete patient profile' });
   }
-})
+});
 
-//Edit Patient Profile
+
+// Edit Patient Profile
 router.put('/myprofile', authenticateToken, async (req, res) => {
   const { firstname, lastname, place_of_birth, date_of_birth, sex, height, weight, occupation, address, phone, complaint } = req.body;
 
   try {
+    // Fetch current patient data
+    const currentPatient = await prisma.patient.findUnique({
+      where: { userId: req.user.id },
+      include: { physician: true }
+    });
+
+    if (!currentPatient) {
+      return res.status(404).json({ error: 'Patient profile not found' });
+    }
+
+    // Prepare the updated data
+    const updatedData = {
+      firstname,
+      lastname,
+      place_of_birth,
+      date_of_birth: new Date(date_of_birth),
+      sex,
+      height: parseInt(height, 10),
+      weight: parseInt(weight, 10),
+      occupation,
+      address,
+      phone,
+      complaint,
+    };
+
+    // Update the patient profile
     const updatedPatient = await prisma.patient.update({
       where: { userId: req.user.id },
-      data: {
-        firstname,
-        lastname,
-        place_of_birth,
-        date_of_birth: new Date(date_of_birth),
-        sex,
-        height: parseInt(height, 10),
-        weight: parseInt(weight, 10),
-        occupation,
-        address,
-        phone,
-        complaint,
-      }
+      data: updatedData
     });
+
+    const physician = currentPatient.physician;
+
+    // Check what fields were changed and create notifications
+    const changes = [];
+    for (const [key, value] of Object.entries(updatedData)) {
+      if (currentPatient[key] !== value) {
+        changes.push(key);
+      }
+    }
+
+    // Notification messages for specific changes
+    const changeMessages = {
+      firstname: `Patient profile edited: ${currentPatient.firstname} ${currentPatient.lastname} changed their name.`,
+      lastname: `Patient profile edited: ${currentPatient.firstname} ${currentPatient.lastname} changed their name.`,
+      place_of_birth: `Patient profile edited: ${currentPatient.firstname} ${currentPatient.lastname} changed their place of birth.`,
+      occupation: `Patient profile edited: ${currentPatient.firstname} ${currentPatient.lastname} changed their occupation.`,
+      address: `Patient profile edited: ${currentPatient.firstname} ${currentPatient.lastname} changed their address.`,
+      complaint: `Patient profile edited: ${currentPatient.firstname} ${currentPatient.lastname} changed their complaint.`,
+    };
+
+    // Create notifications based on changes
+    let specificNotificationSent = false;
+    for (const change of changes) {
```

Comment:
if a user updates several things at once, do we want to send several notifications at once or one notification that groups all the changes?